oauth2-server/src/main/java/com/clouway/oauth2/IssueNewTokenActivity.java

```diff
@@ -5,28 +5,42 @@
 import com.clouway.oauth2.authorization.Authorization;
 import com.clouway.oauth2.authorization.ClientAuthorizationRepository;
 import com.clouway.oauth2.client.Client;
+import com.clouway.oauth2.client.ClientKeyStore;
+import com.clouway.oauth2.jws.Pem;
+import com.clouway.oauth2.jws.Pem.Block;
+import com.clouway.oauth2.jwt.Jwt.Header;
 import com.clouway.oauth2.token.BearerToken;
 import com.clouway.oauth2.token.GrantType;
 import com.clouway.oauth2.token.TokenResponse;
 import com.clouway.oauth2.token.Tokens;
 import com.clouway.oauth2.user.IdentityFinder;
 import com.google.common.base.Optional;
 
+import java.security.SecureRandom;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import static com.clouway.oauth2.idtoken.IdTokenBuilder.newIdToken;
+import static com.clouway.oauth2.jwt.Jwt.ClaimSet.newClaimSet;
+
 /**
  * IssueNewTokenActivity is representing the activity which is performed for issuing of new token.
  *
  * @author Miroslav Genov (miroslav.genov@clouway.com)
  */
 class IssueNewTokenActivity implements ClientActivity {
-
   private final Tokens tokens;
   private final IdentityFinder identityFinder;
   private final ClientAuthorizationRepository clientAuthorizationRepository;
+  private final ClientKeyStore keyStore;
+  private final String clientUrl;
```

Comment:
I've though about this and the additional clientUrl option could be skipped by using the Host header from the request.

The following line:
```java
String host = request.header("Host"); //case for name need to be checked: host Host, HOST
```
should return the target host which was called by the client (which is the iss property actually)